### PR TITLE
Deploy a Docker image using Travis

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Addresses one part of #502. Instead of having the user build the Docker locally every time, we want to push a new Docker image on a tagged commit. In this way, the user can simply download the latest Docker image.

We piggyback off https://github.com/vanvalenlab/deepcell-tf/blob/ad2b019fa912a7d417964429e4c4142abd0d3175/.travis.yml#L72.

**How did you implement your changes**

Add a new entry to the `jobs` section of `.travis.yml` called `docker_deploy` which handles pushing to Docker Hub.

**Remaining issues**

We won't be able to find potential bugs until we push an actual tagged commit. Additionally, some of the following issues may need to be resolved with a 2nd PR:

- We'll need to add our Docker username and password as `$DOCKER_USERNAME` and `$DOCKER_PASSWORD`
- Add a Docker Hub badge to `README.md`
- Change documentation instructing user to download from Docker Hub and not build locally
- Currently, we push to Docker on a commit to `master` every time but to PyPI on just tagged commits to `master`. This workflow may need to be adjusted because of the way Docker tagging works.

### CODE REVIEW

**Comment by alex-l-kong** [comment]
I'm requesting a review on this for now, but some aspects may require some more discussion, an official test on a commit to master, or additional Travis/Docker account setup.

**Comment by alex-l-kong** [comment]
Per our discussion today, once this branch is ready to be merged in I'll change the `docker_deploy` job to run on `tag IS present` as opposed to `branch = master`.

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Okay I updated the environmental variables

File: README.md [hunk 1]
```diff
@@ -21,14 +21,14 @@ Then input the command:
...
- - Once it's sucessfully installed, make sure it is running by looking in toolbar for the Docker whale.
+Next, you'll need to set up the Docker image with all of the required dependencies:
+ - First, [download](https://hub.docker.com/?overlay=onboarding) Docker Desktop. 
+ - Once it's sucessfully installed, make sure it is running by looking in toolbar for the Docker whale. 
  - Once it's running, enter the following commands into terminal 
 
 ```
 cd ark-analysis
-docker build -t ark-analysis '.'
+docker pull angelolab/ark-analysis:latest
```
Comment (ngreenwald) [inline README.md:31]: Did you check that this works? If so looks good

**Review by ngreenwald — CHANGES_REQUESTED** [review]

**Review by ngreenwald — APPROVED** [review]

**Comment by ngreenwald** [comment]
Looks good. Can you make a new release to trigger the push?

**Pull request merged**